kong/clustering.lua

```diff
@@ -351,6 +351,24 @@ local function validate_shared_cert()
   end
 end
 
+local function get_shared_cert_expiration_time()
+  local cert = ngx_var.ssl_client_raw_cert
+
+  if not cert then
+    ngx_log(ngx_ERR, "Data Plane failed to present client certificate " ..
+      "during handshake")
+    return ngx_exit(444)
+  end
+
+  cert = assert(openssl_x509.new(cert, "PEM"))
```

Comment:
I think we should save the parsed X509 object instead of parsing it multiple times for the same connection, as parsing is a relatively expensive operation.